File: src/main/java/io/github/dqualizer/dqlang/types/rqa/configuration/loadtest/LoadTestConfiguration.java
package io.github.dqualizer.dqlang.types.rqa.configuration.loadtest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Set;

/**
 * Java class for the general dqualizer loadtest configuration
 */
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class LoadTestConfiguration {

    private String version;
    private String context;
    private String environment;
    @JsonProperty("base_url")
    private String baseURL;
    @JsonProperty("load_tests")
    private Set<LoadTestArtifact> loadTestArtifacts;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/configuration/loadtest/LoadTestConfiguration.kt
package io.github.dqualizer.dqlang.types.rqa.configuration.loadtest

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Class for the general dqualizer loadtest configuration
 */
data class LoadTestConfiguration(
  val version: String? = null,
  val context: String? = null,
  val environment: String? = null,

  @JsonProperty("base_url")
  val baseURL: String? = null,

  @JsonProperty("load_tests")
  val loadTestArtifacts: Set<LoadTestArtifact>? = HashSet()
)

